packages/x-date-pickers/src/DateCalendar/DateCalendar.types.ts

```diff
@@ -102,13 +101,6 @@ export interface DateCalendarProps<TDate>
    * @default The closest valid date using the validation props, except callbacks such as `shouldDisableDate`.
    */
   referenceDate?: TDate;
-  /**
-   * Callback fired when the value changes.
-   * @template TDate
-   * @param {TDate | null} value The new value.
-   * @param {PickerSelectionState | undefined} selectionState Indicates if the date selection is complete.
-   */
-  onChange?: (value: TDate | null, selectionState?: PickerSelectionState) => void;
```

Comment:
Removing to avoid the duplicate declaration of the same method in `useViews`.
Also helps avoid inconsistency when updating one and forgetting to update others. 🙈